front/src/common/Map/Layers/LineSearchLayer.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Source } from 'react-map-gl';
+
+import { RootState } from 'reducers';
+import { Theme } from 'types';
+
+import { MAP_TRACK_SOURCES, MAP_URL } from 'common/Map/const';
+import OrderedLayer from 'common/Map/Layers/OrderedLayer';
+
+interface TracksGeographicProps {
+  colors: Theme;
+  geomType: string;
+  layerOrder?: number;
+}
+
+function LineSearchLayer(props: TracksGeographicProps) {
+  const { colors, geomType, layerOrder } = props;
+  const { infraID } = useSelector((state: RootState) => state.osrdconf);
+  const { lineSearchCode } = useSelector((state: RootState) => state.map);
```

Comment:
```suggestion
  const { infraID } = useSelector(getInfraID);
  const { lineSearchCode } = useSelector(getLineSearchCode);
```
Use selectors (or create a new one if it does not exists)

Reply:
Done !